pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -270,61 +270,86 @@ func (p *PoolManager) initMacMapFromCluster(parentLogger logr.Logger) error {
 	return nil
 }
 
-// forEachManagedVmInterfaceInClusterRunFunction gets all the macs from all the supported interfaces in all the managed cluster vms, and runs
-// a function f on it
-func (p *PoolManager) forEachManagedVmInterfaceInClusterRunFunction(f func(vmFullName string, iface kubevirt.Interface, networks map[string]kubevirt.Network) error) error {
-	logger := log.WithName("forEachManagedVmInterfaceInClusterRunFunction")
-	var result = p.kubeClient.ExtensionsV1beta1().RESTClient().Get().RequestURI("apis/kubevirt.io/v1/virtualmachines").Do(context.TODO())
-	if result.Error() != nil {
-		return result.Error()
-	}
-
-	vms := &kubevirt.VirtualMachineList{}
-	err := result.Into(vms)
-	if err != nil {
-		return err
-	}
+// paginateVmsWithLimit performs a vm list request with pagination, to limit the amount of vms received at a time
+// and prevent taking too much memory.
+func (p *PoolManager) paginateVmsWithLimit(limit int64, vmsFunc func(pods *kubevirt.VirtualMachineList) error) error {
+	continueFlag := ""
+	for {
+		var result = p.kubeClient.ExtensionsV1beta1().RESTClient().Get().RequestURI(fmt.Sprintf("apis/kubevirt.io/v1/virtualmachines?limit=%v&continue=%v", limit, continueFlag)).Do(context.TODO())
```

Comment:
This feels a little fragile, we don't have the limit/continue option at kubevirt client right ?

Reply:
This is the external api support [way](https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks)..
I think once we'll incorporate the kubevirt client way then we'll be able to use the ListOptions ...

Reply:
Ack, can we add a `//FIXME` note about it ?

Reply:
totally, [Done](https://github.com/k8snetworkplumbingwg/kubemacpool/compare/1064582b3be4f87a570056d764f1af7099fc7536..5bccedba8228960095a047dc893efbf87b2dc465)